website/blog/2024-03-21-using-your-private-git-repository-context-for-completion/index.md

```diff
@@ -0,0 +1,146 @@
+
+```console
+icy@Icys-Ubuntu:~$ docker run -it --gpus all -p 8080:8080 -v $HOME/.tabby:/data tabbyml/tabby serve --model StarCoder-1B --device cuda
+2024-03-21T16:16:47.189632Z  INFO tabby::serve: crates/tabby/src/serve.rs:118: Starting server, this might take a few minutes...
+2024-03-21T16:16:47.190764Z  INFO tabby::services::code: crates/tabby/src/services/code.rs:53: Index is ready, enabling server...    
+ggml_init_cublas: GGML_CUDA_FORCE_MMQ:   no
+ggml_init_cublas: CUDA_USE_TENSOR_CORES: yes
+ggml_init_cublas: found 1 CUDA devices:
+  Device 0: NVIDIA GeForce RTX 4090, compute capability 8.9, VMM: yes
+2024-03-21T16:16:52.464116Z  INFO tabby::routes: crates/tabby/src/routes/mod.rs:35: Listening at 0.0.0.0:8080
+```
+
+Notably, the line `Index is ready, enabling server...` signifies that the server has been successfully launched with the constructed index.
+
+### Automated Scheduler Execution (Optional)
+
+By excluding the `--now` flag from the `scheduler` command, you can configure the scheduler to run automatically at intervals, typically every 10 minutes by default.
+
+To integrate the scheduler with the server, a Docker Compose file can be created as shown below:
+
+```yaml
+version: '3.5'
+name: tabby
+services:
+  serve:
+    restart: always
+    image: tabbyml/tabby
+    command: serve --model StarCoder-1B --device cuda
+    ports:
+      - "8080:8080"
+    volumes:
+      - "~/.tabby:/data"
+    deploy:
+      resources:
+        reservations:
+          devices:
+            - driver: nvidia
+              count: 1
+              capabilities: [gpu]
+  scheduler:
+    restart: always
+    image: tabbyml/tabby
+    entrypoint: /opt/tabby/bin/tabby-cpu
+    command: scheduler
+    volumes:
+      - "~/.tabby:/data"
+```
+
+## Verifying Indexing Results
+
+To confirm that the code completion is effectively utilizing the built index, you can employ the code search feature to validate the indexing process:
+
+1. Access the Swagger UI page: `http://localhost:8080/swagger-ui/#/v1beta/search`.
+2. Click on the `Try it out` button, and input the query parameter `q`, such as `get`.
+3. Click the `Execute` button to trigger the search and see if there are any indexed symbols.
+
+Alternatively, if you have utilized the code completion with the constructed index, you can examine the server log located in `~/.tabby/events` to inspect how the prompt is enhanced during code completion.
+
+## Additional Information on Tabby Enterprise Edition
+
+Starting from version v0.9.0, Tabby Enterprise Edition offers a web UI to manage your git repository contexts. Additionally, a scheduler job management system has been integrated, streamlining the process of monitoring scheduler job statuses. With these enhancements, you can save a lot of effort in maintaining yaml config files and docker compose configurations. Furthermore, users can easily monitor visualized indexing results through the built-in code browser.
```

Comment:
mention that in 0.10.0 we'll support connect directly with github.

Reply:
Updated in 8b5e1e04abfb2ee6ca6d06c523c90059383de006